tests/snmp/test_snmp_queue.py

```diff
@@ -62,3 +74,24 @@ def test_snmp_queues(duthosts, enum_rand_one_per_hwsku_hostname, localhost, cred
             intf = alias_port_name_map[v['name']]
             if intf in q_interfaces and 'queues' not in v:
                 pytest.fail("port %s does not have queue counters" % v['name'])
+            # Check if queue index in QUEUE table in config_db
+            # is present in SNMP result
+            if intf in q_interfaces:
+                for queue_idx in q_interfaces[intf]:
+                    # queue_idx starts with 0, queue_idx in OID starts with 1
+                    # Increment queue_idx by 1 to form the right OID.
+                    snmp_q_idx = int(queue_idx) + 1
+                    if str(snmp_q_idx) not in v['queues'][direction_type]:
+                        pytest.fail("Expected queue index %d not present in \
+                                     SNMP result for interface %s" % (snmp_q_idx, v['name']))
+            # compare number of unicast queues in CLI to the number of queue indexes in
+            # SNMP result
+            if 'queues' in v:
+                if port_name_to_ns[intf]:
+                    show_cli = 'show queue counters -n {} {} | grep "UC" | wc -l'.format(port_name_to_ns[intf], intf)
```

Comment:
>show queue counters [](http://example.com/codeflow?start=32&length=19)

There may be a corner case that `show_cli` has counters for this intf, but snmp has no. Then you should fail the testcase.

Reply:
Is it expected that all interfaces in interfaces mib, will have queue counters associated with it, with or without queue configuration?

Reply:
if `show queue counters` has EthernetN, but snmp has 0 queues on EthernetN, this will pass, but the expectation is failing.

Reply:
Updated to make sure that all interfaces should have queue counters in SNMP result